contrib/aws/internal/eventbridge/eventbridge.go

```diff
@@ -0,0 +1,100 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package eventbridge
+
+import (
+	"encoding/json"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/eventbridge"
+	"github.com/aws/aws-sdk-go-v2/service/eventbridge/types"
+	"github.com/aws/smithy-go/middleware"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"strconv"
+	"time"
+)
+
+const (
+	datadogKey      = "_datadog"
+	startTimeKey    = "x-datadog-start-time"
+	resourceNameKey = "x-datadog-resource-name"
+	maxSizeBytes    = 256 * 1024 // 256 KB
+)
+
+func EnrichOperation(span tracer.Span, in middleware.InitializeInput, operation string) {
+	switch operation {
+	case "PutEvents":
+		handlePutEvents(span, in)
+	}
+}
+
+func handlePutEvents(span tracer.Span, in middleware.InitializeInput) {
+	params, ok := in.Parameters.(*eventbridge.PutEventsInput)
+	if !ok {
+		log.Debug("Unable to read PutEvents params")
+		return
+	}
+
+	for i := range params.Entries {
+		injectTraceContext(span, &params.Entries[i])
+	}
+}
+
+func injectTraceContext(span tracer.Span, entryPtr *types.PutEventsRequestEntry) {
+	if entryPtr == nil {
+		return
+	}
+
+	carrier := tracer.TextMapCarrier{}
```

Comment:
I think you have an opportunity here to do something clever like you did in the java and dotnet tracers. Instead of unmarshalling the whole json blob, what if you just inspect it as a string. Then, you can just remove the trailing `}`, insert our trace context key, then re-add the `}`.

You actually have the chance to do this in two locations. You could json marshall the carrier containing the trace context. Then, for each Entry in the loop, add the bus name and start time manually as strings/bytes.

Doing this trick with the `details` object as a whole is also possible.